### ISSUE TITLE
Unable to import bucket in to module resource

### ISSUE THREAD

**Issue by craigafinch** [terraform-google-modules/terraform-google-cloud-storage#147]
### TL;DR

I am unable to import an existing bucket into this module using standard Terraform import syntax. I get this error: `The given key does not identify an element in this collection value.`

### Expected behavior

Successful import into existing module. 

### Observed behavior

```
terragrunt import 'module.cloud_storage_buckets.google_storage_bucket.buckets["my-bucket-name"]' my-bucket-name       
module.cloud_storage_buckets.google_storage_bucket.buckets["my-bucket-name"]: Importing from ID "my-bucket-name"...
module.cloud_storage_buckets.google_storage_bucket.buckets["my-bucket-name"]: Import prepared!
 Prepared google_storage_bucket for import
module.cloud_storage_buckets.google_storage_bucket.buckets["my-bucket-name"]: Refreshing state... [id=my-bucket-name]

╷
│ Error: Invalid index
│ 
│   on .terraform/modules/cloud_storage_buckets/main.tf line 28, in locals:
│   28:   buckets_list = [for name in var.names : google_storage_bucket.buckets[name]]
│     ├────────────────
│     │ google_storage_bucket.buckets is object with 6 attributes
│ 
│ The given key does not identify an element in this collection value.
╵

ERRO[0002] 1 error occurred:
 * exit status 1
```

### Terraform Configuration

```hcl
variable "project" {}
variable "region" {}
variable "environment" {}
variable "bucket_names" {
  type = list(string)
}

bucket_names = [
  "existing-bucket-1",
  "existing-bucket-2",
  "my-bucket-name",
]

provider "google" {
  project = var.project
  region  = var.region
}

module "cloud_storage_buckets" {
  source  = "terraform-google-modules/cloud-storage/google"
  version = "3.1.0"

  project_id = var.project
  prefix     = var.environment
  location   = "us"
  names      = var.bucket_names
  storage_class = "MULTI_REGIONAL"

  storage_admins = [
    "${var.environment}-compute@${var.project}.iam.gserviceaccount.com",
  ]
}
```

### Terraform Version

```sh
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v3.84.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

### Additional information

I expected import commands to work with this module as described in the Terraform docs: https://www.terraform.io/docs/cli/commands/import.html

I also tried this command but got the same error:
```
terragrunt import 'module.cloud_storage_buckets.google_storage_bucket.buckets' my-bucket-name       
```

I don't know whether this is a bug in this module or an issue with the documentation of this module. Is there a different way to import existing buckets into this module?

**Comment by morgante** [comment]
Can you try importing with *Terraform*? We don't guarantee compatibility with Terragrunt.

**Comment by bharathkkb** [comment]
This might be a core bug https://github.com/hashicorp/terraform/issues/28188

**Comment by bharathkkb** [comment]
>  I don't think this bug is related to the core bug. 

@craigafinch I think its the same bug, since we use that local exclusively for output and from that bug `If there are outputs that refer to a specific element of a resource with a for_each statement, terraform import will fail after importing the first resource of the collection.`.

Could you try a local fork of the module with the outputs and those locals commented out? You can later switch back to the upstream registry after import.
